php-frontend/src/main/java/org/sonar/php/utils/collections/ListUtils.java

```diff
@@ -37,6 +37,19 @@ private ListUtils() {
     return concatenatedList;
   }
 
+  @SafeVarargs
+  public static <T> List<T> merge(Iterable<? extends T>... iterables) {
+    List<T> concatenatedList = new ArrayList<>();
+    for (Iterable<? extends T> iterable : iterables) {
+      iterable.forEach(e -> {
+        if (!concatenatedList.contains(e)) {
+          concatenatedList.add(e);
+        }
+      });
+    }
+    return concatenatedList;
```

Comment:
The complexity of this method looks quite bad because `List.contains` has a linear complexity.